examples/scripts/smart-coffee-machine-client.js

```diff
@@ -17,46 +17,50 @@
-        let waterLevel = await thing.readProperty("availableResourceLevel", { uriVariables: { id: "water" } });
+        const waterLevel = await (
+            await thing.readProperty("availableResourceLevel", { uriVariables: { id: "water" } })
+        ).value();
         log("waterLevel value after change is:", waterLevel);
         // This can also be seen in allAvailableResources property
-        allAvailableResources = await thing.readProperty("allAvailableResources");
+        allAvailableResources = await (await thing.readProperty("allAvailableResources")).value();
         log("allAvailableResources value after change is:", allAvailableResources);
         // It's also possible to set a client-side handler for observable properties
-        thing.observeProperty("maintenanceNeeded", (data) => {
-            log("maintenanceNeeded property has changed! New value is:", data);
-        });
+        /* thing.observeProperty("maintenanceNeeded", async (data) => {
+            log("maintenanceNeeded property has changed! New value is:", data.value());
+        }); */
         // Now let's make 3 cups of latte!
-        let makeCoffee = await thing.invokeAction("makeDrink", undefined, {
+        const makeCoffee = await thing.invokeAction("makeDrink", undefined, {
             uriVariables: { drinkId: "latte", size: "l", quantity: 3 },
         });
-        if (makeCoffee["result"]) {
-            log("Enjoy your drink!", makeCoffee);
+        const makeCoffeep = await makeCoffee.value();
+        if (makeCoffeep.result) {
+            log("Enjoy your drink!", makeCoffeep);
         } else {
-            log("Failed making your drink:", makeCoffee);
+            log("Failed making your drink:", makeCoffeep);
         }
         // See how allAvailableResources property value has changed
-        allAvailableResources = await thing.readProperty("allAvailableResources");
+        allAvailableResources = await (await thing.readProperty("allAvailableResources")).value();
         log("allAvailableResources value is:", allAvailableResources);
         // Let's add a scheduled task
-        let scheduledTask = await thing.invokeAction("setSchedule", {
+        const scheduledTask = await thing.invokeAction("setSchedule", {
             drinkId: "espresso",
             size: "m",
             quantity: 2,
             time: "10:00",
             mode: "everyday",
         });
-        log(scheduledTask["message"], scheduledTask);
+        const scheduledTaskp = await scheduledTask.value();
+        log(scheduledTaskp.message, scheduledTaskp);
         // See how it has been added to the schedules property
-        let schedules = await thing.readProperty("schedules");
+        const schedules = await await (await thing.readProperty("schedules")).value();
         log("schedules value: ", schedules);
         // Let's set up a handler for outOfResource event
         thing.subscribeEvent("outOfResource", (data) => {
```

Comment:
*Warning* `data` is a `InteractionOutput` I would change the code of this callback to actually parse the event value (e.g. `log("outOfResource event",await data.value())`)

Reply:
Thanks for spotting this. Should be resolved by https://github.com/eclipse/thingweb.node-wot/pull/669/commits/c58f25259160b61ecdb6f7fcb1af71e731204877.